### PR TITLE
Add schedules

### PR DESCRIPTION
**Pull request by spetravic**
This PR adds a few schedules that can be used for adjusting parameters during training.

### CODE REVIEW

**Review by singhblom — APPROVED** [review]
Nice.

**Comment by tgolsson** [comment]
Too late to the party; but "get_last_value" is a *terrible* name for the current value... ;-)

**Comment by spetravic** [comment]
@tgolsson I wanted to keep it consistent with get_last_lr() used by Pytorch in their learning rate schedulers because according to them it returns the last computed value. Of course I can rename it :D

**Comment by tgolsson** [comment]
My personal opinion is that we shouldn't copy other people's bad practice. :P I think just `value` or `current` would make a lot more sense! `last` has connotations of "the end", i.e, the value it'd be when the schedule ends. I'd be happy with a property too, as it makes it clear it's cheap but still immutable.
```
@property
def value(self):
    return self._current_val
```
